specs/altair/sync-protocol.md

```diff
@@ -74,6 +107,18 @@ class LightClientUpdate(Container):
     fork_version: Version
 ```
 
+### `OptimisticLightClientUpdate`
+
+```python
+class OptimisticLightClientUpdate(Container):
```

Comment:
With this new "update", then we could probably simplify `LightClientUpdate` to be an update that always has finalized header. It would simplify a lot of the sync logic.

Reply:
`next_sync_committee` should always come from `attested_header`. That was the case before #2746 and is also how the tests are set up. #2806 restores the previous behaviour.

And yes, over this libp2p protocol, `LightClientUpdate` will mostly include a `finalized_header`, unless there is a period of extended non-finality and the `finalized_checkpoint` is so far in the past that the `current_sync_committee` / `next_sync_committee` have to be followed in an optimistic way anyway.

Reply:
After going over this once more, I have to retract the first part of my comment. You are indeed right that `next_sync_committee` for finalized updates are indeed specified as being rooted in the finalized checkpoint; I got confused by the renames in the most recent updates. There was no regression.

Reply:
Related PR: https://github.com/ethereum/consensus-specs/pull/2805

Reply:
A simplification of `LightClientUpdate` to always include a `finalized_header` could be problematic if there is non-finality for an entire sync committee period.

- If `next_sync_committee` is rooted in `finalized_header`, it would become stale and there would not be a way to advance to the next sync committee period anymore.
- If `next_sync_committee` is rooted in `attested_header`, there is no easy way to verify that the path from the `finalized_header` all the way to the current head has been correctly followed.

By only including a `finalized_header` if it is sufficiently recent, this problem is avoided by the current sync protocol design.

Agree that such simplifications would be nice, though. Also agree that it is probably more appropriate to be fixed outside of the work in this PR here.